src/main/java/baseball/number/BaseballNumber.java

```diff
@@ -0,0 +1,108 @@
+package baseball.number;
+
+import camp.nextstep.edu.missionutils.Randoms;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * 야구게임에 필요한 3자리 수를 관리하는 클래스입니다.
+ * 같은 BaseballNumber 클래스끼리 수를 비교할 수 있습니다.
+ */
+public class BaseballNumber {
+    int[] values = {1, 1, 1};
+
+    /**
+     * 사용자의 입력이 규칙에 맞는지 확인합니다.
+     * 1. 입력은 3자리여야 합니다.
+     * 2. 각 자릿수는 1이상 9이하의 숫자여야 합니다.
+     * 3. 각 자릿수의 숫자는 서로 겹치지 않아야 합니다.
+     * @param input 3자리 숫자 형태의 문자열
+     * @return 사용자의 입력이 규칙에 맞는지 여부
+     */
+    private boolean validateInput(String input) {
+        if (input.length() != 3) {
+            return false;
+        }
+
+        Set<Integer> set = new HashSet<>();
+        for (int i = 0; i < 3; i++) {
+            if (!Character.isDigit(input.charAt(i))) {
+                return false;
+            }
+            int value = Character.getNumericValue(input.charAt(i));
+            if (value < 1 || value > 9) {
+                return false;
+            }
+            if (set.contains(value)) {
+                return false;
+            }
+            set.add(value);
+        }
```

Comment:
한 반복문 안에서 3가지 일을 하고 있네요.

- 숫자인지 확인
- 각 숫자의 범위 확인
- 중복된 숫자가 있는지 확인

for문 내에서 하는 일들을 각각 분리하면 메소드로 추출하기도 편합니다.

```suggestion
        for (int i = 0; i < 3; i++) {
            if (!Character.isDigit(input.charAt(i))) {
                return false;
            }
        }
        
        for (int i = 0; i < 3; i++) {
            int value = Character.getNumericValue(input.charAt(i));
            if (value < 1 || value > 9) {
                return false;
            }
        }
        
        for (int i = 0; i < 3; i++) {
            if (set.contains(value)) {
                return false;
            }
            set.add(value);
        }
```

하는 일의 양은 똑같으니 성능 면에서도 크게 차이가 없습니다. :)